### PR TITLE
Devcvl 268 - [FE] - Single Asset (View)

### PR DESCRIPTION
**Pull request by pracheebapate**

### CODE REVIEW

File: src/pages/assets/single/Single.js [hunk 1]
```diff
@@ -59,17 +60,26 @@ const Page = () => {
...
         };
         fetch();
     }, [collection, tokenId]);
 
+    useEffect( () => {
+        const fetch= async() => {
+
+            if(collection && assetId) {
+
+                 const query = new Moralis.Query('CvlNfts');
```
Comment (shakil-adv) [inline src/pages/assets/single/Single.js:73]: you can use `useMoralisQuery` to do the same

File: src/pages/assets/single/Single.js [hunk 2]
```diff
@@ -124,8 +134,11 @@ const Page = () => {
         blockchain: collection ? collection.get("blockchain") : "",
         network: collection ? collection.get("network") : "",
         standard: collection ? collection.get("standard") : "",
+        collectionName: collection ? collection.get("name") : "",
     };
 
+    const collectionLink = collection ? `/collection/${generateURL(nft.collectionName)}`: '';
```
Comment (shakil-adv) [inline src/pages/assets/single/Single.js:140]: does `generateURL()` function handles bypassing duplicate names by adding suffix?

File: src/pages/assets/single/Single.js (hunk 1 above)
Comment (pracheebapate) [reply src/pages/assets/single/Single.js]: Done.

File: src/pages/assets/single/Single.js (hunk 2 above)
Comment (pracheebapate) [reply src/pages/assets/single/Single.js:146]: generateURL() works fine as collection name duplication is handled at create collection. Suppose collection name already exists then V1,V2..Vn is appended to the collection name. 

example:
collection names
Numbers
numbers V1
NumBERs V2
NUMBERS V3
for-ss-2022-05-17_17-14

File: src/pages/assets/single/Single.js [hunk 3]
```diff
@@ -284,9 +298,30 @@ const Page = () => {
...
+                        xl={12}
+                    >
+                        <Button
+                            type="primary"
+                            onClick={handleEdit}
+                        >
+                            Edit
+                        </Button>
+                    </Col>
+                </Row> }
```
[the remark this answers is not in the document]
Comment (pracheebapate) [reply src/pages/assets/single/Single.js:330]: @kyawthetkt I will change it to Only asset owner has to see the Edit button. Thanks

[the remark this answers is not in the document]
Comment (kyawthetkt) [reply src/pages/assets/single/Single.js:330]: @pracheebapate you could leave it as I have to adjust on this page.

**Review by kyawthetkt — APPROVED** [review]

File: src/pages/assets/single/Single.js (hunk 2 above)
Comment (shakil-adv) [reply src/pages/assets/single/Single.js:146]: Okay, thanks for the clarification

**Review by shakil-adv — APPROVED** [review]

**Review by jibrail-idris — APPROVED** [review]

**Pull request merged**